### PR TITLE
[tests/callbacks] refactor with BoringModel

### PR DESCRIPTION
**Pull request by ydcjeff**
## What does this PR do?

Part of #4501 
New of #4517 

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [ ] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests? 
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:  

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified; _Bugfixes should be including in bug-fix release milestones (m.f.X) and features should be included in (m.X.b) releases._
 

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by teddykoker — APPROVED** [review]
Awesome :)

**Review by tchaton — APPROVED** [review]
LGTM !

File: tests/callbacks/test_callbacks.py [hunk 1]
```diff
@@ -51,6 +53,46 @@ def test_trainer_callback_system(torch_save):
...
+    assert callback_mock.on_after_backward.call_count == limit_train_batches
+    assert callback_mock.on_before_zero_grad.call_count == limit_train_batches
+    assert callback_mock.on_batch_end.call_count == limit_train_batches
+    assert callback_mock.on_train_batch_end.call_count == limit_train_batches
+    assert callback_mock.on_save_checkpoint.call_count == 1
+    assert callback_mock.on_epoch_end.call_count == 1
+    assert callback_mock.on_train_epoch_end.call_count == 1
+    assert callback_mock.on_train_end.call_count == 1
+    assert callback_mock.on_fit_end.call_count == 1
+    assert callback_mock.teardown.call_count == 1
```
Comment (Borda) [inline tests/callbacks/test_callbacks.py:85]: use mock with explicit calls, see
https://github.com/PyTorchLightning/pytorch-lightning/blob/0b653b89baf1a36623532d93cb646679d4c8c080/tests/callbacks/test_callbacks.py#L54-L102

**Review by Borda — CHANGES_REQUESTED** [review]

File: tests/callbacks/test_callbacks.py (hunk 1 above)
Comment (ydcjeff) [reply tests/callbacks/test_callbacks.py:85]: you mean to call with arguments?

**Comment by rohitgr7** [comment]
just a suggestion:
here we check which hooks are called when we do `trainer.fit` and `trainer.test`, should we add tests to check which hooks shouldn't be called? for eg: `on_pretrain_routine_start` and `on_pretrain_routine_end` are called when doing trainer.test(), which I assume is incorrect.

**Comment by rohitgr7** [comment]
@ydcjeff updating the base to release/1.2-dev. pls rebase it.

**Comment by ydcjeff** [comment]
Okay, I messed up with merge master and rebase release/1.2-dev. will create a new one.

**Comment by awaelchli** [comment]
I recommend using --onto option when rebasing, or at least I remember that worked when I last had to do it :)

**Comment by Borda** [comment]
> I recommend using --onto option when rebasing, or at least I remember that worked when I last had to do it :)

yes, but looking to the commit history here, the changes are already inside his commits...

**Pull request closed without merging**